File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right = 0, len(height) - 1
        answer = 0  # max area
        while left < right:
            container_h = min(height[left], height[right]) 
            area = container_h * (right - left)
            answer = max(answer, area)  # update the max area
            if height[left] <= height[right]:
                left += 1
            else:
                right -= 1
        
        return answer

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l = 0, r = height.size() - 1, ans = 0;
        
        while (l < r) {
            int area = min(height[l], height[r]) * (r - l);
            ans = max(area, ans);
            if (height[l] <= height[r]) l++;
            else r--;
        }
        
        return ans;
    }
};